EssentialsDiscord/src/main/java/net/essentialsx/discord/util/DiscordUtil.java

```diff
@@ -60,46 +62,51 @@ public static WebhookClient getWebhookClient(long id, String token, OkHttpClient
      *
-     * @return A future which completes with the webhook by the given name in the given channel, if present, otherwise null.
+     * @return A future which completes with the webhook by the given name in the given channel, or null
+     * if the bot lacks the proper permissions.
      */
-    public static CompletableFuture<Webhook> getAndCleanWebhooks(final TextChannel channel, final String webhookName) {
-        final Member self = channel.getGuild().getSelfMember();
+    public static CompletableFuture<Webhook> getOrCreateWebhook(final TextChannel channel, final String webhookName) {
+        if (!channel.getGuild().getSelfMember().hasPermission(channel, Permission.MANAGE_WEBHOOKS)) {
+            return CompletableFuture.completedFuture(null);
+        }
 
         final CompletableFuture<Webhook> future = new CompletableFuture<>();
-        final Consumer<List<Webhook>> consumer = webhooks -> {
-            boolean foundWebhook = false;
+        channel.retrieveWebhooks().queue(webhooks -> {
             for (final Webhook webhook : webhooks) {
-                if (webhook.getName().equalsIgnoreCase(webhookName)) {
-                    if (foundWebhook || !webhook.getChannel().equals(channel)) {
-                        ACTIVE_WEBHOOKS.remove(webhook.getId());
-                        webhook.delete().reason("EssX Webhook Cleanup").queue();
-                        continue;
-                    }
+                if (webhook.getName().equals(webhookName)) {
                     ACTIVE_WEBHOOKS.addIfAbsent(webhook.getId());
                     future.complete(webhook);
-                    foundWebhook = true;
+                    return;
                 }
             }
+            createWebhook(channel, webhookName).thenAccept(future::complete);
+        });
+        return future;
+    }
 
-            if (!foundWebhook) {
-                future.complete(null);
-            }
-        };
-
-        if (self.hasPermission(Permission.MANAGE_WEBHOOKS)) {
-            channel.getGuild().retrieveWebhooks().queue(consumer);
-        } else if (self.hasPermission(channel, Permission.MANAGE_WEBHOOKS)) {
-            channel.retrieveWebhooks().queue(consumer);
-        } else {
-            return CompletableFuture.completedFuture(null);
+    /**
+     * Cleans up unused webhooks from channels that no longer require an advanced relay.
+     *
+     * @param guild       The guild which to preform the cleanup in.
+     * @param webhookName The name of the webhook to scan for.
+     */
+    public static void cleanWebhooks(final Guild guild, String webhookName) {
+        if (!guild.getSelfMember().hasPermission(Permission.MANAGE_WEBHOOKS)) {
+            return;
         }
 
-        return future;
+        guild.retrieveWebhooks().queue(webhooks -> {
+            for (final Webhook webhook : webhooks) {
+                if (webhook.getName().equalsIgnoreCase(webhookName) && !ACTIVE_WEBHOOKS.contains(webhook.getId())) {
```

Comment:
Pretty sure this will break any setups with multiple EssX Discord bots on a single guild?